src/bytes.rs

```diff
@@ -507,6 +507,28 @@ impl Bytes {
         self.truncate(0);
     }
 
+    /// Try to convert self into `BytesMut`.
+    ///
+    /// If `self` is unique, this will succeed and return a `BytesMut` with
+    /// the contents of `self` without copying. If `self` is not unique, this
+    /// will fail and return self.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use bytes::{Bytes, BytesMut};
+    ///
+    /// let bytes = Bytes::from(b"hello".to_vec());
+    /// assert_eq!(bytes.try_into_mut(), Ok(BytesMut::from(&b"hello"[..])));
+    /// ```
+    pub fn try_into_mut(self) -> std::result::Result<BytesMut, Bytes> {
+        if self.is_unique() {
+            Ok(BytesMut::from_vec(self.into()))
```

Comment:
This is probably unavoidable right? Or are the slices of bytes aware of their parents?

Reply:
The `BytesMut` type is also able to reference only part of the allocation. So, it's not unavoidable. (But it will be rather involved.)

Reply:
Ha I read the code and I understand, so essentially if the user did a slice, dropped the other references and then calls `try_into_mut` then it will shift left the data so it can fit into a Vec. But we don't really need this shift since the `BytesMut` could take the same pointer with the offset and length.

Reply:
Re-reading this, `ptr::copy` offers no guarantee that it won't copy even if `ptr == buf` (at least I could not find it). So we should treat it as if it was always copying (or add an if clause).